yb-voyager/src/cp/ybaeon/ybaeon.go

```diff
@@ -0,0 +1,453 @@
+func (ybaeon *YBAeon) createAndSendUpdateRowCountEvent(events []*controlPlane.BaseUpdateRowCountEvent) {
+	timestamp := time.Now().Format(time.RFC3339)
+
+	var tableMetricsList []controlPlane.TableMetrics
+
+	for _, event := range events {
+		metrics := controlPlane.TableMetrics{
+			MigrationUUID:       event.MigrationUUID,
+			TableName:           event.TableName,
+			SchemaName:          strings.Join(event.SchemaNames, "|"),
+			MigrationPhase:      controlPlane.MIGRATION_PHASE_MAP[event.EventType],
+			Status:              controlPlane.UPDATE_ROW_COUNT_STATUS_STR_TO_INT[event.Status],
+			CountLiveRows:       event.CompletedRowCount,
+			CountTotalRows:      event.TotalRowCount,
+			InvocationTimestamp: timestamp,
+		}
+		tableMetricsList = append(tableMetricsList, metrics)
+	}
+
+	select {
+	case ybaeon.rowCountUpdateEventChan <- tableMetricsList:
+		ybaeon.waitGroup.Add(1)
+	default:
+		log.Warnf("Could not publish row count update event to YB-Aeon (channel full) %v", tableMetricsList)
+	}
+}
+
+// getInvocationSequence retrieves the next invocation sequence for a migration phase
+func (ybaeon *YBAeon) getInvocationSequence(migrationUUID uuid.UUID, phase int) (int, error) {
+	// Optimization: If already initialized, just increment
+	if ybaeon.latestInvocationSequence > 0 {
+		ybaeon.latestInvocationSequence += 1
+		return ybaeon.latestInvocationSequence, nil
+	}
+
+	// Query YB-Aeon API for max sequence
+	maxSeq, err := ybaeon.getMaxSequenceFromAPI(migrationUUID, phase)
+	if err != nil {
+		log.Warnf("Failed to get max sequence from YB-Aeon API: %v. Falling back to sequence 1", err)
+		ybaeon.latestInvocationSequence = 1
+		return ybaeon.latestInvocationSequence, nil
+	}
+
+	ybaeon.latestInvocationSequence = maxSeq + 1
+	return ybaeon.latestInvocationSequence, nil
+}
+
+// getMaxSequenceFromAPI calls the GET /voyager/migrations/{migrationId}/phases/{phase}/latest-sequence endpoint
+// HTTP client handles retries automatically with exponential backoff
+func (ybaeon *YBAeon) getMaxSequenceFromAPI(migrationUUID uuid.UUID, phase int) (int, error) {
+	path := fmt.Sprintf("/api/public/v1/accounts/%s/projects/%s/clusters/%s/voyager/migrations/%s/phases/%d/latest-sequence",
+		ybaeon.config.AccountID, ybaeon.config.ProjectID, ybaeon.config.ClusterID, migrationUUID.String(), phase)
+
+	var response struct {
+		Data struct {
+			LatestSequence int `json:"latest_sequence"`
+		} `json:"data"`
+	}
+
+	statusCode, err := ybaeon.httpClient.Get(context.Background(), path, &response)
+
+	// Special case: 404 means no prior data for this phase
+	if statusCode == 404 {
+		log.Infof("No prior data for phase %d (404), using sequence 0 (first run)", phase)
+		return 0, nil
+	}
+
+	if err != nil {
+		// After all retries failed, fallback to sequence 0 to allow migration to proceed
+		log.Warnf("Failed to get max sequence from YB-Aeon: %v. Falling back to sequence 0", err)
+		return 0, nil
+	}
```

Comment:
it should not be the responsibility of `getMaxSequenceFromAPI` to decide whether it's okay to ignore the error or not, it should return the error; caller should decide what do with the error.